addons/binding/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/handler/FeedHandler.java

```diff
@@ -0,0 +1,290 @@
+     * {@link #SUPPORTED_FEED_FORMATS}.
+     *
+     * @param numberOfEntries - number of {@link SyndEntry} or all entries, if they are less than this number
+     * @param feed
+     * @return <code>String</code> containing the XML data or <code>null</code> if XML can not be created
+     */
+    private String getFeedContent(SyndFeed feed, int numberOfEntries) {
+        List<SyndEntry> allEntries = feed.getEntries();
+
+        if (allEntries.size() > numberOfEntries) {
+            /*
+             * The entries are stored in the SyndFeed object in the following order -
+             * the newest entry has index 0. The order is determined from the time the entry was posted, not the
+             * published time of the entry.
+             */
+            allEntries = allEntries.subList(0, numberOfEntries);
+            feed.setEntries(allEntries);
+            logger.debug("Content will be generated from the first {} feed entries.", numberOfEntries);
+        }
+        return getFeedContent(feed);
+    }
+
+    /**
+     * Creates a String with the XML representation for the given {@link SyndFeed} in one of the specified formats
+     * {@link #SUPPORTED_FEED_FORMATS}.This method sets the Feed format as well.
+     *
+     * @param feed
+     * @return <code>String</code> containing the XML data or <code>null</code> if XML is invalid for the selected
+     *         {@link #outputFeedFormat}
+     */
+    private String getFeedContent(SyndFeed feed) {
+        feed.setFeedType(outputFeedFormat);
+
+        if (outputFeedFormat.equals("rss_0.91N") || outputFeedFormat.equals("rss_0.91U")) {
+            // RSS 0.91 has required language attribute. Default value is assigned, because if this value is missing,
+            // theconversion to RSS 0.91 will fail
+            if (feed.getLanguage() == null) {
+                feed.setLanguage("en-us");
+            }
+        }
+
+        SyndFeedOutput outputBuilder = new SyndFeedOutput();
+        String content = null;
+        try {
+            content = outputBuilder.outputString(feed);
+        } catch (Exception e) {
+            logger.error("The XML representation for the feed could not be created. {}", e.getMessage());
+        }
+        return content;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            if (channelUID.getId().equals(FEED_CHANNEL)) {
+                refreshFeedState();
+            } else {
+                logger.debug("Command received for an unknown channel: {}", channelUID.getId());
+            }
+        } else {
+            logger.debug("Command {} is not supported for channel: {}. Supported command: REFRESH", command,
+                    channelUID.getId());
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (refreshTask.isDone()) {
+            logger.error("Refresh task is not terminated properly");
```

Comment:
Isn't it perfectly valid that the job might just be running and is not yet done? If so, you should not log an error.